src/dectris/neggia/data/H5Superblock.cpp

```diff
@@ -50,12 +56,18 @@ ResolvedPath H5Superblock::resolveV0(const H5Path& path) {
 
 ResolvedPath H5Superblock::resolveV2(const H5Path& path) {
     // verify header information
-    int version = (int)fileAddress()[8];
-    assert(version == 2);
     int offsetSize = (int)fileAddress()[9];
     assert(offsetSize == 8);
     int offsetLength = (int)fileAddress()[10];
     assert(offsetLength == 8);
+    if (version() == 3) {
+        // we need to check that the file is not open for write access
+        uint8_t fileConsistencyFlags = (uint8_t)fileAddress()[11];
+        if ((fileConsistencyFlags & 5) > 0)
+            throw std::runtime_error(
+                    "file with superblock version 3 may not be open for write "
```

Comment:
Maybe simply: "file opened for write access"? Superblock version not relevant here?